src/SmoFacade/Listener.cs

```diff
@@ -1,137 +1,134 @@
-﻿namespace AgDatabaseMove.SmoFacade
-{
-  using System;
-  using System.Collections.Generic;
-  using System.Data.SqlClient;
-  using System.Linq;
-  using System.Net;
-  using System.Threading.Tasks;
-
-
-  internal interface IListener : IDisposable
-  {
-    /// <summary>
-    ///   The primary instance at the time of construction.
-    /// </summary>
-    Server Primary { get; set; }
-
-    /// <summary>
-    ///   The secondary instances at the time of construction.
-    /// </summary>
-    IEnumerable<Server> Secondaries { get; set; }
-
-    /// <summary>
-    ///   The availability group associated with the supplied listener.
-    /// </summary>
-    AvailabilityGroup AvailabilityGroup { get; set; }
-
-    /// <summary>
-    ///   Executes the provided action on each availability group server instance.
-    ///   This may use one or more threads to execute in parallel.
-    /// </summary>
-    void ForEachAgInstance(Action<Server, AvailabilityGroup> action);
-
-    /// <summary>
-    ///   Executes the provided action on each availability group server instance.
-    ///   This may use one or more threads to execute in parallel.
-    /// </summary>
-    void ForEachAgInstance(Action<Server> action);
-  }
-
-  internal class Listener : IListener
-  {
-    public Listener(string connectionString)
-    {
-      /*
-       * We initially connect an availability group instance by way of a listener name. This creates a different connection and
-       * SMO server object from connecting directly to the primary instance. Having different SMO server objects results in the
-       * data cached by SMO getting out of sync between the two and forces us to refresh the objects regularly. This should
-       * prevent us from having to do the refreshes by only using SMO objects that connect directly to the server. The initial
-       * connection to find those server names though is done through the listener and that instance is thrown away in this
-       * constructor so it won't be used again.
-       */
-
-      var connectionStringBuilder = new SqlConnectionStringBuilder(connectionString);
-      connectionStringBuilder.InitialCatalog = "master";
```

Comment:
This is an implementation detail of how the facade uses the connection string given to connect to secondaries and shouldn't be made the responsibility of the caller. If the database name is given in the connection string it will fail to connect to the secondaries.

Reply:
I was unsure why this was here.
> If the database name is given in the connection string it will fail to connect to the secondaries.

Actually? Why?

Reply:
The secondary nodes don't have to be readable. If you attempt to connect to a database on a secondary that is marked as non-readable the connection will fail. Our use case in particular will have non readable secondaries.